contracts/voting-verifier/src/contract/execute.rs

```diff
@@ -157,6 +168,117 @@ pub fn verify_messages(
     }))
 }
 
+pub fn verify_message_with_payload(
+    deps: DepsMut,
+    env: Env,
+    messages_with_payloads: Vec<MessageWithPayload>,
+) -> Result<Response, ContractError> {
+    if messages_with_payloads.is_empty() {
+        return Err(report!(ContractError::EmptyMessages));
+    }
+
+    let config = CONFIG.load(deps.storage).expect("failed to load config");
+
+    let messages = messages_with_payloads.try_map(|message_with_payload| {
+        // Message needs to be to ITS Hub
+        if message_with_payload.message.destination_chain.as_ref()
+            != config.axelar_chain_name.as_ref()
+            || message_with_payload.message.destination_address.as_str()
+                != config.its_hub_address.as_str()
+        {
+            return Err(ContractError::InvalidMessage.into());
```

Comment:
probably want to include the actual invalid message here

Reply:
Done